src/Microsoft.IdentityModel.Tokens/SecurityToken.cs

```diff
@@ -64,5 +64,10 @@ public abstract class SecurityToken
         /// This must be overridden to get the time when this <see cref="SecurityToken"/> is no longer Valid.
         /// </summary>
         public abstract DateTime ValidTo { get; }
+
+        /// <summary>
+        /// Gets the original raw data of this instance when it was created.
+        /// </summary>
+        public string RawToken { get; protected internal set; }
```

Comment:
A custom SecurityToken will have to expose an internal method so the custom SecurityTokenHandler will be able to set it.
Could we code this as:
```cs
public virtual string RawToken { get;} = string.Empty;
```
or
```cs
public string RawToken {get;set;}
```
Then each SecurityTokenHandler will know how to set the RawToken, they can have their own contract.

Reply:
This makes sense, changed to a public setter as suggested so it is consistent with, for example, the SigningKey property.